astropy/table/table.py

```diff
@@ -1778,89 +1778,57 @@ def _make_index_row_display_table(self, index_row_name):
         else:
             return self
 
-    @deprecated(
-        "6.1",
-        message="""show_in_notebook() is deprecated as of 6.1 and to create
-         interactive tables it is recommended to use dedicated tools like:
-         - https://github.com/bloomberg/ipydatagrid
-         - https://docs.bokeh.org/en/latest/docs/user_guide/interaction/widgets.html#datatable
-         - https://dash.plotly.com/datatable""",
-    )
-    def show_in_notebook(
-        self,
-        tableid=None,
-        css=None,
-        display_length=50,
-        table_class="astropy-default",
-        show_row_index="idx",
-    ):
-        """Render the table in HTML and show it in the IPython notebook.
+    def show_in_notebook(self, *, backend="ipydatagrid", **kwargs):
+        """Render the table in HTML and show it in the Jupyter notebook.
+
+        .. note:: The method API was modified in v7.0 to include a ``backend``
+           argument and require only keyword arguments.
 
         Parameters
         ----------
-        tableid : str or None
-            An html ID tag for the table.  Default is ``table{id}-XXX``, where
-            id is the unique integer id of the table object, id(self), and XXX
-            is a random number to avoid conflicts when printing the same table
-            multiple times.
-        table_class : str or None
-            A string with a list of HTML classes used to style the table.
-            The special default string ('astropy-default') means that the string
-            will be retrieved from the configuration item
-            ``astropy.table.default_notebook_table_class``. Note that these
-            table classes may make use of bootstrap, as this is loaded with the
-            notebook.  See `this page <https://getbootstrap.com/css/#tables>`_
-            for the list of classes.
-        css : str
-            A valid CSS string declaring the formatting for the table. Defaults
-            to ``astropy.table.jsviewer.DEFAULT_CSS_NB``.
-        display_length : int, optional
-            Number or rows to show. Defaults to 50.
-        show_row_index : str or False
-            If this does not evaluate to False, a column with the given name
-            will be added to the version of the table that gets displayed.
-            This new column shows the index of the row in the table itself,
-            even when the displayed table is re-sorted by another column. Note
-            that if a column with this name already exists, this option will be
-            ignored. Defaults to "idx".
+        backend : {"ipydatagrid", "classic"}
+            Backend to use for rendering (default="ipydatagrid"). The "classic" backend
+            is deprecated since v6.1.
+
+        **kwargs : dict, optional
+            Keyword arguments as accepted by desired backend.
```

Comment:
Add a note here like "See `astropy.table.notebook_backends` for the available backends and their respective keyword arguments."

As always I'm not sure if I have that sphinx link to the API docs right but you get the idea.